Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1471,6 +1471,12 @@ void LocalFrame::showResourceMonitoringError()
     document->addConsoleMessage(MessageSource::ContentBlocker, MessageLevel::Error, "Frame was unloaded because its network usage exceeded the limit."_s);
 }
 
+void LocalFrame::reportResourceMonitoringWarning()
+{
+    if (RefPtr document = this->document())
+        document->addConsoleMessage(MessageSource::ContentBlocker, MessageLevel::Warning, "Frame's network usage exceeded the limit."_s);
```

Comment:
Can protectedDocument() be used here?

Reply:
No, Basuke's code is correct. We don't want to call `protectedDocument()` since we already have a `RefPtr` from the null check above.